streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -2155,12 +2161,18 @@ class ZStream[-R, +E, +A] private[stream] (private[stream] val structure: ZStrea
    * leaving the remainder `R0`.
    */
   final def provideSomeM[R0, E1 >: E](env: ZIO[R0, E1, R])(implicit ev: NeedsEnv[R]): ZStream[R0, E1, A] =
-    provideSomeManaged(env.toManaged_)
+    ZStream.managed {
```

Comment:
I think we can deprecate `provideSomeM` as it's the same as `provideLayer(zio.toLayer)`, right?

Reply:
Yes. The only issue is what if the `R` type is not a subtype of `Has` so you can't call `toLayer`. But it could always be replaced with `env.flatMap(zio.provideSome)`.

Reply:
Only the A parameter needs to be, not R, right?

Reply:
The `A` type of the `ZLayer` needs to be which would be the `R` type of the `ZIO` that the layer is being provided to.